go/logic/topology_recovery.go

```diff
@@ -1411,6 +1455,46 @@ func checkAndRecoverGenericProblem(analysisEntry inst.ReplicationAnalysis, candi
 	return false, nil, nil
 }
 
+// checkAndRecoverDeadGroupMemberWithReplicas checks whether action needs to be taken for an analysis involving a dead
+// replication group member, and takes the action if applicable. Notice that under our view of the world, a primary
+// replication group member is akin to a master in traditional async/semisync replication; whereas secondary group
+// members are akin to intermediate masters. Considering also that a failed group member can always be considered as a
+// secondary (even if it was primary, the group should have detected its failure and elected a new primary), then
+// failure of a group member with replicas is akin to failure of an intermediate master.
+func checkAndRecoverDeadGroupMemberWithReplicas(analysisEntry inst.ReplicationAnalysis, candidateInstanceKey *inst.InstanceKey, forceInstanceRecovery bool, skipProcesses bool) (bool, *TopologyRecovery, error) {
+	// Don't proceed with recovery unless it was forced or automatic intermediate source recovery is enabled.
+	// We consider failed group members akin to failed intermediate masters, so we re-use the configuration for
+	// intermediates.
+	if !(forceInstanceRecovery || analysisEntry.ClusterDetails.HasAutomatedIntermediateMasterRecovery) {
+		return false, nil, nil
+	}
+	// Try to record the recovery. It it fails to be recorded, it because it is already being dealt with.
+	topologyRecovery, err := AttemptRecoveryRegistration(&analysisEntry, !forceInstanceRecovery, !forceInstanceRecovery)
+	if err != nil {
+		return false, nil, err
+	}
+	// Proceed with recovery
+	recoverDeadReplicationGroupMemberCounter.Inc(1)
+
+	recoveredToGroupMember, err := RecoverDeadReplicationGroupMemberWithReplicas(topologyRecovery, skipProcesses)
+
+	if recoveredToGroupMember != nil {
+		// success
+		recoverDeadReplicationGroupMemberSuccessCounter.Inc(1)
+
+		if !skipProcesses {
+			// Execute post failover processes
+			topologyRecovery.SuccessorKey = &recoveredToGroupMember.Key
+			topologyRecovery.SuccessorAlias = recoveredToGroupMember.InstanceAlias
+			// For the same reasons that were mentioned above, we re-use the post intermediate master fail-over hooks
+			executeProcesses(config.Config.PostIntermediateMasterFailoverProcesses, "PostIntermediateMasterFailoverProcesses", topologyRecovery, false)
```

Comment:
I don't run Group Replication myself, but I think it can be debatable whether it is correct to run `PostIntermediateMasterFailoverProcesses`. For now, let's keep it at that, but I predict that someone in the future will argue against this.

Reply:
For now, our use case does not seem to require different hooks for these. If the need arises (or someone comes knocking at your door asking for it) I'd be happy to add them as separate hooks.